app/components/UI/Ramp/hooks/useRegions.ts

```diff
@@ -1,10 +1,30 @@
-import { useCallback, useEffect, useMemo } from 'react';
+import { useCallback, useEffect, useMemo, useState } from 'react';
 import { useNavigation, useRoute } from '@react-navigation/native';
 import { isEqual } from 'lodash';
 import Routes from '../../../../constants/navigation/Routes';
 import { useRampSDK } from '../sdk';
 import { Region } from '../types';
 import useSDKMethod from './useSDKMethod';
+import { Country, State } from '@consensys/on-ramp-sdk';
+
+export const isCountry = (region: Country | State | null): region is Country =>
+  (region as Country).states !== undefined;
+
+const findDetectedRegion = (regions: (Country | State)[]): Region | null => {
+  for (const region of regions) {
+    if (region.detected) {
+      if (isCountry(region) && region.states.length > 0) {
+        const detectedState = region.states.find((state) => state.detected);
+        if (detectedState) {
+          return detectedState as Region;
+        }
+      } else {
+        return region as Region;
+      }
+    }
+  }
+  return null;
+};
```

Comment:
nit: we can refactor this to use recursion  
```suggestion
const findDetectedRegion = (regions: (Country | State)[]): Region | null => {
  const detectedRegion = regions.find((region) => region.detected);
  if (!detectedRegion) return null;
  if (isCountry(detectedRegion) && detectedRegion.states.length > 0) {
    return findDetectedRegion(detectedRegion.states);
  }
  return detectedRegion as Region;
};
```